llmfoundry/data/finetuning/collator.py

```diff
@@ -235,6 +235,7 @@ def __init__(
         target_prompts: str = 'none',
         allow_pad_trimming: bool = False,
         batch_metadata: Optional[Dict[str, Any]] = None,
+        is_profiling: bool = False,
```

Comment:
let's rename this flag. Would prefer a flag that describes the functionality of what happens rather than an overall state so that the flag makes sense if you use the collator independently. 

Ideally, if the below comment makes sense, we can just use the allow_pad_trimming flag

Reply:
also since this is only supported for decoder_only_format, please add a checks that errors if this is not the case

Reply:
Renamed and added support for _process_and_batch_encoder_decoder